### ISSUE TITLE
Logging v2 - fluentbit pods fail to reach the fluentd service 3.14.0

### ISSUE THREAD

**Issue by izaac** [rancher/rancher#34772]
**Rancher Server Setup**
- Rancher version: `v2.5-head (09/16/2021)` _ad21cff_
- Installation option: Helm HA
   - If Helm Chart, Kubernetes Cluster and version: RKE1
- Proxy/Cert Details: rancher generated / no proxy

**Information about the Cluster**
- Kubernetes version: `1.20.10-rancher1-1`
- Cluster Type: Downstream
   - If downstream, what type of cluster?: Custom Docker `20.10.7` on EC2

**Describe the bug**

Rancher Logging chart version: 3.14.0 

The fluentbit pods of the `rancher-logging-rke-aggregator` daemonset show an error that they can't connect to the fluentd service.

This works on chart

```
[2021/09/16 18:53:12] [ warn] [engine] failed to flush chunk '1-1631818381.605162503.flb', retry in 25 seconds: task_id=3, input=tail.0 > output=forward.0 (out_id=0)
[2021/09/16 18:53:14] [ warn] [net] getaddrinfo(host='rancher-logging-fluentd.cattle-logging-system.svc', err=4): Domain name not found
[2021/09/16 18:53:14] [error] [output:forward:forward.0] no upstream connections available
[2021/09/16 18:53:14] [ warn] [engine] failed to flush chunk '1-1631818390.64151719.flb', retry in 7 seconds: task_id=5, input=tail.0 > output=forward.0 (out_id=0)
```

**To Reproduce**

- Install the 3.14.0 Rancher logging chart
- Deploy the [logging examples](https://github.com/paynejacob/rancher-logging-examples) (`make install` with the proper KUBECONFIG)
- Check the logs of the fluentbit pods in the `rancher-logging-rke-aggregator` daemonset

**Result**

```
[2021/09/16 18:53:12] [ warn] [engine] failed to flush chunk '1-1631818381.605162503.flb', retry in 25 seconds: task_id=3, input=tail.0 > output=forward.0 (out_id=0)
[2021/09/16 18:53:14] [ warn] [net] getaddrinfo(host='rancher-logging-fluentd.cattle-logging-system.svc', err=4): Domain name not found
[2021/09/16 18:53:14] [error] [output:forward:forward.0] no upstream connections available
[2021/09/16 18:53:14] [ warn] [engine] failed to flush chunk '1-1631818390.64151719.flb', retry in 7 seconds: task_id=5, input=tail.0 > output=forward.0 (out_id=0)
```

**Expected Result**
Logs flowing to the backend

**Screenshots**

**Additional context**
This works with chart version `3.9.400+up3.9.4`
This was checked on Ubuntu 18.04 and RHEL 8.4

**Comment by paynejacob** [comment]
I can reproduce this on RKE v1.21.5.  It looks like dns lookups are not working properly.

```
/ # nslookup rancher-logging-fluentd.cattle-logging-system.svc
Server:         10.43.0.10
Address:        10.43.0.10:53

** server can't find rancher-logging-fluentd.cattle-logging-system.svc: NXDOMAIN
```

```
/ # nslookup rancher-logging-fluentd.cattle-logging-system.svc.cluster.local
Server:         10.43.0.10
Address:        10.43.0.10:53

Name:   rancher-logging-fluentd.cattle-logging-system.svc.cluster.local
Address: 10.43.229.129
```

```
/ # cat /etc/resolv.conf
nameserver 10.43.0.10
search cattle-logging-system.svc.cluster.local svc.cluster.local cluster.local
options ndots:5
/ #
```

**Comment by paynejacob** [comment]
I should add this worked fine on a 1.19 rke cluster

**Comment by izaac** [comment]
Rancher version: `v2.5-head (10/01/2021)` _5b2469f_
Chart version: `v3.14.0`
Fluent-bit version: `1.8.7`

I am still hitting this issue on an RKE1 cluster with k8s v1.20.11-rancher1-1

```
[2021/10/01 18:47:17] [ warn] [net] getaddrinfo(host='rancher-logging-fluentd.cattle-logging-system.svc', err=4): Domain name not found
[2021/10/01 18:47:17] [error] [output:forward:forward.0] no upstream connections available
[2021/10/01 18:47:17] [ warn] [engine] failed to flush chunk '1-1633114036.47762838.flb', retry in 11 seconds: task_id=1, input=tail.0 > output=forward.0 (out_id=0)
```

**Comment by izaac** [comment]
Rancher version `2.5-head (10/06/2021)` _6b04f9b_

I validated the new fluentbit image was `v1.6.10` but we ran into an issue too because the configcheck workloads wasn't applying new or updating configurations after creating ClusterOutputs/ClusterFlows.

**Comment by ronhorton** [comment]
**PASS** Verified `FIXED` in `2.5-head Commit ID bea1658`
1. Created a cluster with 1.20 k8s version
2. installed logging `rancher-logging:3.15.0`
3. Configured logging
4. Confirmed index passed through to logging target - in this case elasticsearch

Result: ` fluentbit pods reach the fluentd service 3.15.0`

**Comment by jazzl0ver** [comment]
Facing the same issue on K8s installed from Kubespray release-2.19 and Rancher 2.6.6. Rancher Logging version is 100.1.2+up3.17.4 installed from ClusterTools:
```
[2022/07/27 15:43:24] [ warn] [net] getaddrinfo(host='rancher-logging-root-fluentd.cattle-logging-system.svc.cluster.local', err=1): DNS server returned answer with no data
```
Any ideas how to fix?

The host is resolvable:
```
# Run kubectl commands inside here
# e.g. kubectl get all
> nslookup rancher-logging-root-fluentd.cattle-logging-system.svc.cluster.local
Server:         169.254.25.10
Address:        169.254.25.10:53

Name:   rancher-logging-root-fluentd.cattle-logging-system.svc.cluster.local
Address: 10.233.47.219
```
PS fluentbit image version is 1.8.15
